libs/unity/library/Tests/Runtime/PeerConnectionTests.cs

```diff
@@ -28,25 +73,118 @@ public IEnumerator SimplePeerConnection()
 
             // Initialize
-            var initializedEvent = new ManualResetEventSlim(initialState: false);
-            pc.OnInitialized.AddListener(() => initializedEvent.Set());
+            yield return InitializeAndWait(pc);
+            UnityEngine.Object.Destroy(go);
+        }
+
+        [UnityTest]
+        public IEnumerator EnableAndDisable()
+        {
+            // Create the component
+            var go = new GameObject("test_go");
+            go.SetActive(false); // prevent auto-activation of components
+            var pc = go.AddComponent<PeerConnection>();
             Assert.IsNull(pc.Peer);
-            pc.InitializeAsync().Wait();
+            for (int i = 0; i < 2; ++i)
+            {
+                yield return InitializeAndWait(pc);
+                yield return ShutdownAndCheckEvent(pc);
+            }
+
+            UnityEngine.Object.Destroy(go);
+        }
+
+        [UnityTest]
+        public IEnumerator EnableAndDisableWithTracks()
+        {
+            var pc1_go = new GameObject("pc1");
+            pc1_go.SetActive(false); // prevent auto-activation of components
+            var pc1 = pc1_go.AddComponent<PeerConnection>();
+            var pc2_go = new GameObject("pc2");
+            pc2_go.SetActive(false); // prevent auto-activation of components
+            var pc2 = pc2_go.AddComponent<PeerConnection>();
 
-            // Wait a frame so that the Unity event OnInitialized can propagate
-            yield return null;
+            // Create the signaler
+            var sig_go = new GameObject("signaler");
+            var sig = sig_go.AddComponent<LocalOnlySignaler>();
+            sig.Peer1 = pc1;
+            sig.Peer2 = pc2;
 
-            // Check the event was raised
-            Assert.IsTrue(initializedEvent.Wait(millisecondsTimeout: 50000));
-            Assert.IsNotNull(pc.Peer);
+            // Create the video source on peer #1
+            VideoTrackSource source1 = pc1_go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver1 = pc1_go.AddComponent<VideoReceiver>();
+            MediaLine ml1 = pc1.AddMediaLine(MediaKind.Video);
+            ml1.SenderTrackName = "video_track_1";
+            ml1.Source = source1;
+            ml1.Receiver = receiver1;
+
+            // Create the video source on peer #2
+            VideoTrackSource source2 = pc2_go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver2 = pc2_go.AddComponent<VideoReceiver>();
+            MediaLine ml2 = pc2.AddMediaLine(MediaKind.Video);
+            ml2.SenderTrackName = "video_track_2";
+            ml2.Source = source2;
+            ml2.Receiver = receiver2;
+
+            // Init/quit twice.
+            for (int i = 0; i < 2; ++i)
+            {
+                // Initialize
+                var init1 = InitializeAndWait(pc1);
+                var init2 = InitializeAndWait(pc2);
+                yield return init1;
+                yield return init2;
```

Comment:
```suggestion
                yield return InitializeAndWait(pc1);
                yield return InitializeAndWait(pc2);
```

Reply:
Other places too; it's unusual to see a `IEnumerator` captured instead of just returning it / yielding it. Although technically I guess there's nothing wrong with this.

Reply:
The idea is to start the two async jobs together then wait for them. Though chances are that in this case they will be run on the same thread, so it doesn't make much difference, I'll go with the two-liner.